src/System.Reflection/tests/CoreCLR/Assembly/AssemblyTests.CoreCLR.cs

```diff
@@ -12,21 +12,18 @@ namespace System.Reflection.Tests
     public class AssemblyTests
     {
         [Fact]
-        public void CurrentAssemblyHasLocation()
+        public void CurrentLocation_HasLocaton()
         {
-            string location = ThisAssembly().Location;
-
-            Assert.NotNull(location);
-            Assert.NotEmpty(location);
-
+            string location = GetExecutingAssembly().Location;
+            Assert.False(string.IsNullOrEmpty(location));
```

Comment:
I've replaced this with `Assert.NotEmpty`. This checks for null explicitly as well as the empty case, differentiating between the two. Same goes for `Assert.Empty`.

See the code [here](https://github.com/xunit/assert.xunit/blob/bbece92b18661cc18ede9d72b28648466f9a1709/CollectionAsserts.cs#L182)